src/backend/distributed/metadata/node_metadata.c

```diff
@@ -573,36 +582,348 @@ master_set_node_property(PG_FUNCTION_ARGS)
+		{
+			distArgumentIndexList = lappend_int(distArgumentIndexList,
+												INVALID_DISTRIBUTION_ARGUMENT_INDEX);
+		}
+		else
+		{
+			distArgumentIndexList = lappend_int(distArgumentIndexList,
+												distributionArgumentIndex);
+		}
+
+		if (colocationIdIsNull)
+		{
+			colocationIdList = lappend_int(colocationIdList,
+										   INVALID_COLOCATION_ID);
+		}
+		else
+		{
+			colocationIdList = lappend_int(colocationIdList, colocationId);
+		}
+	}
+
+	systable_endscan_ordered(pgDistObjectScan);
+	index_close(pgDistObjectIndexRel, AccessShareLock);
+	relation_close(pgDistObjectRel, NoLock);
+
+	char *workerMetadataUpdateCommand =
+		MarkObjectsDistributedCreateCommand(objectAddressList,
+											distArgumentIndexList,
+											colocationIdList);
+	List *commandList = list_make1(workerMetadataUpdateCommand);
+
+	return commandList;
+}
+
+
+/*
+ * UpdatePgDistLocalGroupOnNode updates the pg_dist_local_group on the given node
+ */
+static void
+UpdatePgDistLocalGroupOnNode(WorkerNode *workerNode)
+{
+	/* generate and add the local group id's update query */
+	char *localGroupIdUpdateCommand = LocalGroupIdUpdateCommand(workerNode->groupId);
+
+	List *localGroupIdUpdateCommandList = list_make1(localGroupIdUpdateCommand);
+
+	SendMetadataCommandListToWorkerInCoordinatedTransaction(workerNode->workerName,
+															workerNode->workerPort,
+															CitusExtensionOwnerName(),
+															localGroupIdUpdateCommandList);
+}
+
+
+/*
+ * ClearDistributedTablesFromNode clear (shell) distributed tables from the given node.
+ */
+static void
+ClearDistributedTablesFromNode(WorkerNode *workerNode)
+{
+	List *clearDistributedTablesCommandList = NIL;
+
+	List *detachPartitionCommandList = DetachPartitionCommandList();
+
+	clearDistributedTablesCommandList = list_concat(clearDistributedTablesCommandList,
+													detachPartitionCommandList);
+
+	clearDistributedTablesCommandList = lappend(clearDistributedTablesCommandList,
+												REMOVE_ALL_CLUSTERED_TABLES_ONLY_COMMAND);
+
+	clearDistributedTablesCommandList = list_concat(list_make1(DISABLE_DDL_PROPAGATION),
+													clearDistributedTablesCommandList);
+	clearDistributedTablesCommandList = list_concat(clearDistributedTablesCommandList,
+													list_make1(
+														ENABLE_DDL_PROPAGATION));
+
+	SendMetadataCommandListToWorkerInCoordinatedTransaction(workerNode->workerName,
+															workerNode->workerPort,
+															CitusExtensionOwnerName(),
+															clearDistributedTablesCommandList);
+}
+
+
+/*
+ * ClearDistributedObjectsFromNode clears all the distributed objects, metadata and partition hierarchy from the given node.
+ */
+static void
+ClearDistributedObjectsFromNode(WorkerNode *workerNode)
+{
+	List *clearDistTableInfoCommandList = NIL;
+
+	clearDistTableInfoCommandList = lappend(clearDistTableInfoCommandList,
+											DELETE_ALL_PARTITIONS);
+
+	clearDistTableInfoCommandList = lappend(clearDistTableInfoCommandList,
```

Comment:
Can you please remind me why we had to `DELETE_ALL_SHARDS`, `DELETE_ALL_PLACEMENTS`  etc

Reply:
In case we have incorrect metadata on worker nodes (most probably because of manual update), we won't be able to make metadata on worker nodes completely same with the metadata on coordinator node. We discussed that we can use it also for operational purposes.